path_login.go

```diff
@@ -170,30 +175,69 @@ func (b *azureAuthBackend) verifyClaims(claims *additionalClaims, role *azureRol
 	return nil
 }
 
-func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, resourceGroupName, vmName string, claims *additionalClaims, role *azureRole) error {
+func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, resourceGroupName, vmName string, vmssName string, claims *additionalClaims, role *azureRole) error {
 	// If not checking anything with the resource id, exit early
-	if len(role.BoundResourceGroups) == 0 && len(role.BoundSubscriptionsIDs) == 0 && len(role.BoundLocations) == 0 {
+	if len(role.BoundResourceGroups) == 0 && len(role.BoundSubscriptionsIDs) == 0 && len(role.BoundLocations) == 0 && len(role.BoundScaleSets) == 0 {
 		return nil
 	}
 
-	if subscriptionID == "" || resourceGroupName == "" || vmName == "" {
-		return errors.New("subscription_id, resource_group_name, and vm_name are required")
+	if subscriptionID == "" || resourceGroupName == "" {
+		return errors.New("subscription_id and resource_group_name are required")
 	}
 
-	client := b.provider.ComputeClient(subscriptionID)
-	vm, err := client.Get(ctx, resourceGroupName, vmName, compute.InstanceView)
-	if err != nil {
-		return errwrap.Wrapf("unable to retrieve virtual machine metadata: {{err}}", err)
+	var principalID, location *string
+
+	switch {
+	case vmssName != "" && vmName == "":
```

Comment:
Instead of this, would you mind just adding a switch at the top when both are not empty and error on that case.  I think that would be a little more clear.

Reply:
Will do!

Reply:
@chrishoffman I've just realised I might have misunderstood your intent here. Are you saying the test to check if both vmss_name and vm_name are set should go at the top of `verifyResource` rather than at the top of _this_ switch statement?

Reply:
Nope, this looks good to me.